src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -245,51 +244,43 @@ bool InsertSpecificIterations::decompose(LinearIR& linear_ir,
                                   }
                               });
 
-                std::map<UnifiedLoopInfoPtr, UnifiedLoopInfoPtr> unified_loop_map;
-                auto get_unified_cloned_info = [&unified_loop_map,
-                                                &expression_map](const ExpandedLoopInfoPtr& expanded_loop_info) {
-                    const auto& unified_loop_info = expanded_loop_info->get_unified_loop_info();
-                    if (unified_loop_map.count(unified_loop_info) == 0) {
-                        LoopInfoMap loop_info_map;
-                        // Note: we must clone UnifiedLoopInfo for the cloned ExpandedLoopInfos
-                        auto cloned_info = ov::as_type_ptr<UnifiedLoopInfo>(
-                            unified_loop_info->clone_with_new_expr(expression_map, loop_info_map));
-                        OPENVINO_ASSERT(cloned_info, "cloned info must be UnifiedLoopInfo");
-                        unified_loop_map[unified_loop_info] = cloned_info;
-                    }
-                    OPENVINO_ASSERT(unified_loop_map.count(unified_loop_info),
-                                    "Cloned UnifiedLoopInfo must be cloned at this stage.");
-                    return unified_loop_map[unified_loop_info];
-                };
-
                 // Note: all internal decomposed loops must be also cloned to avoid a situation
                 // when 2 loops with the same ID exist in both specific iterations of the outer loop
+                LoopInfoMap loop_info_map;
                 for (auto it = std::next(decomposed_loop_bounds.first); it != decomposed_loop_bounds.second; ++it) {
                     auto internal_loop_end = ov::as_type_ptr<op::LoopEnd>(it->get()->get_node());
                     if (!internal_loop_end) {
                         continue;
                     }
                     const auto loop_begin = internal_loop_end->get_loop_begin();
-                    auto begin_it = linear_ir.find_after(std::next(decomposed_loop_bounds.first),
-                                                         linear_ir.get_expr_by_node(loop_begin));
-                    OPENVINO_ASSERT(begin_it != linear_ir.cend(),
```

Comment:
Is there any reason on why this assertion was removed? It still seems valid for me